neqo-transport/src/crypto.rs

```diff
@@ -861,115 +861,132 @@ impl CryptoStreams {
+            } => (Some(initial), Some(handshake), Some(application)),
             Self::Handshake {
                 handshake,
                 application,
-            } => (None, Some(handshake), application),
-            Self::ApplicationData { application } => (None, None, application),
+            } => (None, Some(handshake), Some(application)),
+            Self::ApplicationData { application } => (None, None, Some(application)),
         };
         match space {
-            PNSpace::Initial => initial.expect("Initial state dropped!"),
-            PNSpace::Handshake => hs.expect("Handshake state dropped!"),
+            PNSpace::Initial => initial,
+            PNSpace::Handshake => hs,
             PNSpace::ApplicationData => app,
         }
     }
-}
 
-impl IndexMut<PNSpace> for CryptoStreams {
-    fn index_mut(&mut self, space: PNSpace) -> &mut Self::Output {
+    fn get_mut(&mut self, space: PNSpace) -> Option<&mut CryptoStream> {
         let (initial, hs, app) = match self {
             Self::Initial {
                 initial,
                 handshake,
                 application,
-            } => (Some(initial), Some(handshake), application),
+            } => (Some(initial), Some(handshake), Some(application)),
             Self::Handshake {
                 handshake,
                 application,
-            } => (None, Some(handshake), application),
-            Self::ApplicationData { application } => (None, None, application),
+            } => (None, Some(handshake), Some(application)),
+            Self::ApplicationData { application } => (None, None, Some(application)),
         };
         match space {
-            PNSpace::Initial => initial.expect("Initial state dropped!"),
-            PNSpace::Handshake => hs.expect("Handshake state dropped!"),
+            PNSpace::Initial => initial,
+            PNSpace::Handshake => hs,
             PNSpace::ApplicationData => app,
         }
     }
+
+    pub fn get_frame(
+        &mut self,
+        space: PNSpace,
+        remaining: usize,
+    ) -> Option<(Frame, Option<RecoveryToken>)> {
+        if let Some(cs) = self.get_mut(space) {
+            if let Some((offset, data)) = cs.tx.next_bytes() {
+                let (frame, length) = Frame::new_crypto(offset, data, remaining);
+                cs.tx.mark_as_sent(offset, length);
+
+                qdebug!(
+                    "Emitting crypto frame space={}, offset={}, len={}",
+                    space,
+                    offset,
+                    length
+                );
+                Some((
+                    frame,
+                    Some(RecoveryToken::Crypto(CryptoRecoveryToken {
+                        space,
+                        offset,
+                        length,
+                    })),
+                ))
+            } else {
+                None
+            }
+        } else {
+            qerror!("Trying to get frame for dropped space {:?}", space);
```

Comment:
Can you work out how to reach this state?  Because it seems like a panic!() or unreachable!() might be better